### PR TITLE
Refactor fix name usage

### PR DESCRIPTION
**Pull request by BenFordTytherington**
Update and rename some aspects surrounding name usage, and fix a bug with qml_element generation

### CODE REVIEW

File: crates/cxx-qt-gen/src/naming/name.rs [hunk 1]
```diff
@@ -237,6 +237,25 @@ impl Name {
...
+    pub fn require_module_ident(&self) -> Result<&Ident> {
+        if let Some(ident) = self.module_ident() {
+            Ok(ident)
+        } else {
+            Err(Error::new_spanned(
+                self.rust_unqualified(),
+                format!("No Module name for {}!", self.rust_unqualified()),
+            ))
+        }
+    }
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/naming/name.rs:257]: some docs lines on the two methods would be good :-)

File: crates/cxx-qt-gen/src/parser/qobject.rs [hunk 2]
```diff
@@ -127,15 +126,15 @@ impl ParsedQObject {
...
+        name: &Name,
         attrs: &mut Vec<Attribute>,
     ) -> Result<Option<QmlElementMetadata>> {
         // Find if there is a qml_element attribute
         if let Some(attr) = attribute_take_path(attrs, &["qml_element"]) {
             // Extract the name of the qml_element
             let name = match attr.meta {
                 Meta::NameValue(name_value) => expr_to_string(&name_value.value)?,
-                _ => qobject_ident.to_string(),
+                _ => name.cxx_unqualified(),
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/parser/qobject.rs:137]: could add a comment above so it reads
```
// Extract the name of the qml_element
// or fallback to using the C++ class name
```

Comment (BenFordTytherington) [reply crates/cxx-qt-gen/src/parser/qobject.rs:138]: Done

File: crates/cxx-qt-gen/src/naming/name.rs (hunk 1 above)
Comment (BenFordTytherington) [reply crates/cxx-qt-gen/src/naming/name.rs]: Done

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**